### PR TITLE
[KYUUBI #833] Check if `spark.kubernetes.executor.podNamePrefix` is invalid

### LINKED ISSUE #833: Check if `spark.kubernetes.executor.podNamePrefix` is invalid
### 1. Describe the feature

https://github.com/apache/spark/pull/32610

### PR DESCRIPTION
**Pull request by Kwafoor**
### _Why are the changes needed?_
close #833 

### _How was this patch tested?_
- [ ] Add some test cases that check the changes thoroughly including negative and positive cases if possible

- [ ] Add screenshots for manual tests if appropriate

- [x] [Run test](https://kyuubi.apache.org/docs/latest/develop_tools/testing.html#running-tests) locally before make a pull request

### CODE REVIEW

**Comment by ulysses-you** [comment]
cc @yaooqinn @zwangsheng @hddong

File: kyuubi-server/src/main/scala/org/apache/kyuubi/util/Validator.scala [hunk 1]
```diff
@@ -0,0 +1,52 @@
...
+object Validator {
+
+  def validateConf(kyuubiConf: KyuubiConf): Unit = {
+    validateK8sConf(kyuubiConf)
+  }
+
+  private def validateK8sConf(kyuubiConf: KyuubiConf): Unit = {
+    val prefix = kyuubiConf.getOption(KUBERNETES_EXECUTOR_POD_NAME_PREFIX).orNull
+    if (prefix != null && !isValidExecutorPodNamePrefix(prefix)) {
+      throw new IllegalArgumentException(s"'$prefix' " +
```
Comment (ulysses-you) [inline kyuubi-server/src/main/scala/org/apache/kyuubi/util/Validator.scala:31]: KyuubiSQLException

File: kyuubi-server/src/main/scala/org/apache/kyuubi/util/Validator.scala [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.kyuubi.util
+
+import org.apache.kyuubi.config.KyuubiConf
+
+object Validator {
```
Comment (zwangsheng) [inline kyuubi-server/src/main/scala/org/apache/kyuubi/util/Validator.scala:22]: Now kyuubi supports multiple engines. For Spark-related content, add Spark prefix. For example, `Validator => SparkValidator` or the following method is changed to `validateSparkK8sConf`

**Review by zwangsheng — COMMENTED** [review]
LGTM, with one comment

**Comment by yaooqinn** [comment]
we can move the conf validator closer to proc builder of spark, for 2 main reasons,

1. it's unnecessary to do this in KyuubiSessionImpl as it will affect other engine types too
2. a session could successfully establish even if the conf is invalid as it only fails when creating apps.

**Comment by Kwafoor** [comment]
> we can move the conf validator closer to proc builder of spark, for 2 main reasons,
> 
> 1. it's unnecessary to do this in KyuubiSessionImpl as it will affect other engine types too
> 2. a session could successfully establish even if the conf is invalid as it only fails when creating apps.

I agree, so I changed validateConf to ProcBuilder.And is it necessary for other engine types validateConf?

**Comment by yaooqinn** [comment]
> > we can move the conf validator closer to proc builder of spark, for 2 main reasons,
> > 
> > 1. it's unnecessary to do this in KyuubiSessionImpl as it will affect other engine types too
> > 2. a session could successfully establish even if the conf is invalid as it only fails when creating apps.
> 
> I agree, so I changed validateConf to ProcBuilder.And is it necessary for other engine types validateConf?

yes, we can do this separately

**Review by zwangsheng — APPROVED** [review]
LGTM

**Comment by Kwafoor** [comment]
> > > we can move the conf validator closer to proc builder of spark, for 2 main reasons,
> > > 
> > > 1. it's unnecessary to do this in KyuubiSessionImpl as it will affect other engine types too
> > > 2. a session could successfully establish even if the conf is invalid as it only fails when creating apps.
> > 
> > 
> > I agree, so I changed validateConf to ProcBuilder.And is it necessary for other engine types validateConf?
> 
> yes, we can do this separately

Ok.

**Review by wForget — APPROVED** [review]

**Comment by Kwafoor** [comment]
cc @yaooqinn @ulysses-you

**Review by ulysses-you — APPROVED** [review]

**Comment by ulysses-you** [comment]
thanks, merging to master

**Pull request closed without merging**